scripts/install_python_prod_dependencies.py

```diff
@@ -280,15 +299,23 @@ def _rectify_third_party_directory(mismatches):
         return
 
     git_mismatches, pip_mismatches = (
-        utils.partition(mismatches.items(), predicate=_is_git_url_mismatch))
+        utils.partition(
+            mismatches.items(),
+            predicate=_is_git_url_mismatch,
+            enumerated=False
+        )
+    )
 
     for normalized_library_name, versions in git_mismatches:
         requirements_version, directory_version = versions
 
         # The library listed in 'requirements.txt' is not in the
         # 'third_party/python_libs' directory.
         if not directory_version or requirements_version != directory_version:
-            _install_direct_url(normalized_library_name, requirements_version)
+            if requirements_version:
+                _install_direct_url(
+                    normalized_library_name, requirements_version
+                )
```

Comment:
The previous code seems to assume that `requirements_version` is not falsy, so should we assert that here?